File: backend/src/entities/security_product.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "security_products")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub name: String,
    pub category: String,
    pub vendor: String,
    pub model: String,
    pub version: String,
    pub serial_number: Option<String>,
    pub license_type: String,
    pub license_expiry: Option<String>,
    pub management_ip: Option<String>,
    pub deployment_mode: String,
    pub cloud_platform_id: Option<i32>,
    pub machine_room_id: Option<i32>,
    pub provider_id: Option<i32>,
    pub status: String,
    pub features: Option<String>, // JSON array
    pub throughput: Option<String>,
    pub contact_person: String,
    pub contact_phone: String,
    pub remarks: Option<String>,
    pub created_at: String,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(
        belongs_to = "super::cloud_platform_config::Entity",
        from = "Column::CloudPlatformId",
        to = "super::cloud_platform_config::Column::Id"
    )]
    CloudPlatformConfig,
    #[sea_orm(
        belongs_to = "super::machine_room::Entity",
        from = "Column::MachineRoomId",
        to = "super::machine_room::Column::Id"
    )]
    MachineRoom,
    #[sea_orm(
        belongs_to = "super::service_provider::Entity",
        from = "Column::ProviderId",
        to = "super::service_provider::Column::Id"
    )]
    ServiceProvider,
}

impl Related<super::cloud_platform_config::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::CloudPlatformConfig.def()
    }
}

impl Related<super::machine_room::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::MachineRoom.def()
    }
}

impl Related<super::service_provider::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::ServiceProvider.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}

File: apps/web/apps/web-antd/src/api/scan/security-product.ts
import { requestClient } from '#/api/request';

export namespace ScanSecurityProductApi {
  export interface SecurityProduct {
    id?: number; name: string; category: string; vendor: string;
    model: string; version: string; serial_number?: string;
    license_type: string; license_expiry?: string; management_ip?: string;
    deployment_mode: string; cloud_platform_id?: number; machine_room_id?: number;
    provider_id?: number; status: string; features?: string;
    throughput?: string; contact_person: string; contact_phone: string;
    remarks?: string; created_at: string;
  }
}
export function getSecurityProductList() { return requestClient.get<ScanSecurityProductApi.SecurityProduct[]>('/asset-ops/security-products'); }
export function getSecurityProduct(id: number) { return requestClient.get<ScanSecurityProductApi.SecurityProduct>(`/asset-ops/security-products/${id}`); }
export function createSecurityProduct(data: ScanSecurityProductApi.SecurityProduct) { return requestClient.post('/asset-ops/security-products', data); }
export function updateSecurityProduct(id: number, data: ScanSecurityProductApi.SecurityProduct) { return requestClient.put(`/asset-ops/security-products/${id}`, data); }
export function deleteSecurityProduct(id: number) { return requestClient.delete(`/asset-ops/security-products/${id}`); }
